### PR TITLE
Update Travis configuration and add badge to README

### PR DESCRIPTION
**Pull request by lackac**
Related to #60.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,23 @@
 language: elixir
 elixir:
-  - 1.5.1
```
Comment (josevalim) [inline .travis.yml]: We still support 1.5.1, so I would say we still need to have it here.

File: .travis.yml [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
-    - jose.valim@plataformatec.com.br
+    - jose.valim@gmail.com
+    - lackac@lackac.hu
 otp_release:
-  - 18.0
+  - 19.3
+  - 20.3
+  - 21.0
 script:
-  - if [[ $(elixir --version) = *"1.6"* ]]; then mix format --check-formatted; fi
```
Comment (josevalim) [inline .travis.yml]: Therefore we need to bring this back for now.

File: .travis.yml [hunk 3]
```diff
@@ -1,16 +1,23 @@
...
 notifications:
   recipients:
-    - jose.valim@plataformatec.com.br
+    - jose.valim@gmail.com
+    - lackac@lackac.hu
 otp_release:
-  - 18.0
+  - 19.3
+  - 20.3
+  - 21.0
```
Comment (josevalim) [inline .travis.yml]: This is bound to become a probem, it is better to list the Elixir / OTP pairs together.

**Comment by lackac** [comment]
Will that still be the case after 1.7 is released?

File: .travis.yml (hunk 1 above)
Comment (lackac) [reply .travis.yml]: Will that still be the case after 1.7 is released?
What is the rule for choosing the patch level for a supported minor release? i.e. why `1.5.1` and not `1.5.3`?

**Comment by lackac** [comment]
@josevalim I would say that CI works now, but there are some failures on `1.5.3`. I chose the latest releases from each minor branch of Elixir and OTP for the matrix.

**Comment by josevalim** [comment]
@lackac the segmentation fault is a VM bug, so we need to pick another version. The other seem to be timing related, we can probably do something like this to be safer: https://github.com/elixir-lang/elixir/blob/ae8502d8cde107f30c170ef510ce2177d1c442d8/lib/elixir/test/elixir/test_helper.exs#L88

**Comment by lackac** [comment]
@josevalim I couldn't find a working combination of Elixir 1.5 and OTP 18. See https://travis-ci.org/elixir-lang/flow/builds/407973578

All of those except for two are clear segmentation faults. The other two look to be related as they call `:erlang.apply/3` with something that is not a module in the first argument looking to be coming from the wrong memory space: https://travis-ci.org/elixir-lang/flow/jobs/407973581

I think it would be better to drop tests on OTP 18.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**